frontend/src/metabase/xray/components/InsightCard.jsx

```diff
@@ -0,0 +1,368 @@
+import React, { Component } from 'react'
+import {
+    //formatListOfItems,
+    formatTimeWithUnit,
+    //inflect
+} from "metabase/lib/formatting";
+import Icon from "metabase/components/Icon";
+import { Link } from "react-router";
+import Question from "metabase-lib/lib/Question";
+import { TermWithDefinition } from "metabase/components/TermWithDefinition";
+import { t } from 'c-3po'
+
+const InsightText = ({ children }) =>
+    <p className="text-paragraph">
+        {children}
+    </p>
+
+const Feedback = ({ insightType }) =>
+    <div className="flex align-center px1">
+        {t`Was this helpful?`}
+        <div className="ml-auto text-bold">
+            <a className="text-brand-hover" data-metabase-event={`InsightFeedback;${insightType};Yes`}>
+                {t`Yes`}
+            </a>
+            <a className="text-brand-hover ml1" data-metabase-event={`InsightFeedback;${insightType};No`}>
+                {t`No`}
+            </a>
+        </div>
+    </div>
+
+export class NormalRangeInsight extends Component {
+    static insightType = "normal-range"
+    static title = "Normal range of values"
+    static icon = "insight"
+
+    render() {
+        const { lower, upper, features: { model } } = this.props
+        return (
+            <InsightText>
+                Most of the values for { model.display_name || model.name } are between <b>{ lower }</b> and <b>{ upper }</b>.
+            </InsightText>
+        )
+    }
+}
+
+export class NilsInsight extends Component {
+    static insightType = "nils"
+    static title = "Missing data"
+    static icon = "warning"
+
+    render() {
+        const { quality, filter, features: { table } } = this.props
+
+        const viewAllRowsUrl = table && Question.create()
+            .query()
+            // imitate the required hydrated metadata format
+            .setTable({ ...table, database: { id: table.db_id }})
+            .addFilter(filter)
+            .question()
+            .getUrl()
+
+        // construct the question with filter
+        return (
+            <InsightText>
+                You have { quality } missing (null) values in your data.
```

Comment:
👍  if we're in the process of doing i18n we should definitely make sure all new PRs going forward have everything tagged up so we keep making forward progress instead of getting further away from our goal of 100% of things tagged

Reply:
yeah, this should be one of the standard requirements for merging a PR in addition to tests (and instrumentation in case of bigger features like this)